ZelBack/src/services/syncthingService.js

```diff
@@ -96,11 +99,258 @@ async function systemPing(req, res) {
   return res ? res.json(response) : response;
 }
 
+async function getConfigRestartRequired(req, res) {
+  const response = await performRequest('get', '/rest/config/restart-required');
+  return res ? res.json(response) : response;
+}
+
 async function getConfigFolders(req, res) {
-  const response = await performRequest('get', '/rest/config/folders');
+  let { id } = req.params;
+  id = id || req.query.id;
+  let path = '/rest/config/folders';
+  if(id){
+    path = `/rest/config/folders/${id}`;
+  }
+  const response = await performRequest('get', path);
+  return res ? res.json(response) : response;
+}
+
+async function getConfigDevices(req, res) {
+  let { id } = req.params;
+  id = id || req.query.id;
+  let path = '/rest/config/devices';
+  if(id){
+    path = `/rest/config/devices/${id}`;
+  }
+  const response = await performRequest('get', path);
+  return res ? res.json(response) : response;
+}
+
+async function postConfigFolders(req, res) {
+  let { data } = req.params;
+  data = data || req.query.data;
+  if (!data) {
+    throw new Error('data is not provided');
+  }
+  const authorized = await verificationHelper.verifyPrivilege('admin', req);
```

Comment:
authorized has to happen if res is present. if not its always true as fluxos shall be having access to all

Reply:
is there any implementation of this elsewhere? will add it after the tests are done.

Reply:
You can do
`const authorized = res ? await verificationHelper.verifyPrivilege('admin', req) : true;`